websocket-tcp-proxy/src/main.rs

```diff
@@ -100,20 +101,30 @@ fn format_addr_from_url(ip: String, port: String) -> String {
 
 // Big help from https://github.com/HsuJv/axum-websockify
 async fn handle_socket(mut socket: WebSocket, ip: String, port: String) {
-    let addr = format_addr_from_url(ip, port);
-    let server_stream = TcpStream::connect(&addr).await;
-    tracing::info!("Connect to remote {:#?}", server_stream);
+    let addr_str = format_addr_from_url(ip, port);
+    let mut addrs = addr_str.to_socket_addrs().expect("Failed to resolve");
 
-    if server_stream.is_err() {
-        tracing::error!("Connect to remote failed {:#?}", server_stream);
+    if addrs.len() == 0 {
+        tracing::error!("Could not resolve addr {addr_str}");
         let _ = socket
-            .send(Message::Text(format!("{:#?}", server_stream)))
+            .send(Message::Text(format!("Could not resolve addr {addr_str}")))
             .await;
         return;
     }
 
+    let server_stream = addrs.find_map(|addr| {
+        let connection = block_on(TcpStream::connect(&addr));
+        if let Err(error) = &connection {
+            tracing::error!("Could not connect to {addr}: {error}");
+        };
+
+        connection.ok()
+    });
+
     let mut server_stream = server_stream.unwrap();
```

Comment:
this unwrap is bad now because the .is_err case isn't being handled. if the connection fails we panic:

2022-12-19T18:14:26.541204Z ERROR websocket_tcp_proxy: Could not connect to 52.41.246.129:9735: Connection refused (os error 61)
thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value', websocket-tcp-proxy/src/main.rs:124:43
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace